libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -267,6 +267,9 @@ - (void)replayRequestIfRequired:(NSData *)data response:(NSURLResponse *)respons
             NSError *parsingError;
             NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parsingError];
             if (parsingError) {
+                if (data == nil || [data isEqualToData:[[NSData alloc] init]]) {
```

Comment:
Given that `nil` data would zero out integer properties as well, you could simplify this whole conditional to `if (data.length == 0)`.